test/micromamba/test_config.py

```diff
@@ -529,25 +559,55 @@ def test_file_prepend_single_input(self):
         ).splitlines() == ["channels:", "  - flowers"]
 
     def test_file_prepend_multiple_inputs(self):
+        with open(TestConfigModifiers.home_rc_path, "w") as f:
+            f.write("channels:\n  - foo")
+
+        config(
+            "prepend",
+            "channels",
+            "condesc,mambesc",
+            "--file",
+            TestConfigModifiers.home_rc_path,
+        )
         assert (
             config(
-                "prepend",
-                "channels",
-                "condesc",
-                "mambesc",
-                "--file",
-                TestConfigModifiers.home_rc_path,
+                "get", "channels", "--file", TestConfigModifiers.home_rc_path
             ).splitlines()
-            == "Prepend key is invalid or it's not present in file or more than one key was received".splitlines()
+            == "channels:\n  - condesc\n  - mambesc\n  - foo".splitlines()
         )
 
-    def test_file_prepend_invalid_input(self):
+    def test_file_prepend_multiple_keys(self):
+        with open(TestConfigModifiers.home_rc_path, "w") as f:
+            f.write("channels:\n  - foo\ndefault_channels:\n  - bar")
+
+        config(
+            "prepend",
+            "channels",
+            "condesc,mambesc",
+            "default_channels",
+            "condescd,mambescd",
+            "--file",
+            TestConfigModifiers.home_rc_path,
+        )
         assert (
             config(
-                "prepend", "@#A321", "--file", TestConfigModifiers.home_rc_path
+                "get", "channels", "--file", TestConfigModifiers.home_rc_path
             ).splitlines()
-            == "Prepend key is invalid or it's not present in file or more than one key was received".splitlines()
+            == "channels:\n  - condesc\n  - mambesc\n  - foo".splitlines()
         )
+        assert (
+            config(
+                "get", "default_channels", "--file", TestConfigModifiers.home_rc_path
+            ).splitlines()
+            == "default_channels:\n  - condescd\n  - mambescd\n  - bar".splitlines()
+        )
+
+    def test_file_prepend_invalid_input(self):
+        with pytest.raises(subprocess.CalledProcessError):
+            config("prepend", "--file", TestConfigModifiers.home_rc_path)
+
+        with pytest.raises(subprocess.CalledProcessError):
+            config("prepend", "@#A321", "--file", TestConfigModifiers.home_rc_path)
 
     def test_file_append_and_prepend_inputs(self):
         config(
```

Comment:
I think we should add tests for the following cases:

1. Non vector key prepending/appending:
`./micromamba config prepend json true` and `./micromamba config append json true`

2. Vector key appending + non-vector key appending:
`./micromamba config prepend channels wawah json true`

Awesome work! :)
Thank you for this @adriendelsalle. The code is shining!

Reply:
totally agree with the missing tests, will do that ;)
thanks for those kind words and for taking time to review @marimeireles :+1: